CreditManagement/views.py

```diff
@@ -15,44 +16,201 @@ class TransactionListView(ListView):
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(TransactionAlterView, self).get_context_data(*args, **kwargs)
+
+        transaction_obj = self.get_transaction()
+        # get the form
+        context['form'] = self.get_form(transaction=transaction_obj)
+
+        context['redirect'] = self.request.GET.get('redirect')
+        context['chain'] = self.request.GET.get('chain', False)
+
+        # Set the correct template
+        if transaction_obj is None:
+            self.template_name = self.template_add_name
         else:
-            self.context['slot_form'] = UserTransactionForm(request.user)
-        return render(request, self.template_name, self.context)
+            self.template_name = self.template_edit_name
+        return context
 
-    def post(self, request, association_name=None):
-        # Do form shenanigans
-        if association_name:
-            association = Association.objects.get(slug=association_name)
+    def post(self, request):
+        transaction_obj = self.get_transaction()
+        form = self.get_form(transaction=transaction_obj, data=request.POST)
+
+        if form.is_valid():
+            form.save()
+            if transaction_obj is None:
+                messages.add_message(request, messages.SUCCESS, _("Transaction has been added successfully."))
+            else:
+                messages.add_message(request, messages.SUCCESS, _("Transaction has been changed successfully."))
+
+            if 'add_another' in request.POST:
+                url = request.path_info + "?chain=True"
+                if request.GET.get('redirect'):
+                    url += "&redirect=" + request.GET.get('redirect')
+                if request.GET.get('amount'):
+                    url += "&amount=" + request.GET.get('amount')
+                return HttpResponseRedirect(url)
+
+            return HttpResponseRedirect(request.GET.get('redirect', request.path_info))
+
+        context = self.get_context_data()
+        context['form'] = form
+        return render(request, self.template_name, context)
+
+
+class TransactionUserView(TransactionAlterView):
+    def get_form(self, transaction=None, data=None):
+        return UserTransactionForm(self.request.user,
+                                   initial_from_get=self.request.GET,
+                                   data=data,
+                                   instance=transaction)
+
+    def check_access_permission(self, request):
+        transaction_id = request.GET.get('id', None)
+
+        if transaction_id is not None:
+            # get the instance
+            t_order = get_object_or_404(PendingTransaction, pk=transaction_id)
+            if t_order.source_user != request.user:
+                return HttpResponseForbidden("You do not have access to this transaction")
+            if t_order.confirm_moment <= timezone.now():
+                return HttpResponseForbidden("This transaction can no longer be altered")
```

Comment:
> The HTTPResponseForbidden yields a 403 page where the error is presented.

You might think that but I'm pretty sure that that's not the case.. You see the message only because you have DEBUG=True in your settings. On the production server though the DEBUG setting is False which will ensure that only the literal text '403 Forbidden' will be shown to the user (this is mentioned in the Django docs..). It's easy to create a custom 403/404 page, by adding 403.html/404.html to the root template folder. However I don't want that as it's totally unnecessary. A user should never end up on a 403 or 404 page via UI interactions, so only via other means and I don't want views that are devoted to guiding people that are not using the 'official UI'